extension/data.ts

```diff
@@ -199,190 +188,152 @@ class RcxDict {
+                reason: '',
+              } as Deinflection;
               if (have[newWord] !== undefined) {
-                o = r[have[newWord]];
+                o = possibleDeinflections[have[newWord]];
                 o.type |= rule.typeMask >> 8;
-
                 continue;
               }
-              have[newWord] = r.length;
-              if (r[i].reason.length) {
+              have[newWord] = possibleDeinflections.length;
+              if (possibleDeinflections[i].reason?.length) {
                 o.reason =
-                  this.difReasons[rule.reasonIndex] + ' &lt; ' + r[i].reason;
+                  this.difReasons[rule.reasonIndex] +
+                  ' &lt; ' +
+                  possibleDeinflections[i].reason;
               } else {
                 o.reason = this.difReasons[rule.reasonIndex];
               }
               o.type = rule.typeMask >> 8;
               o.word = newWord;
-              r.push(o);
+              possibleDeinflections.push(o);
             }
           }
         }
       }
-    } while (++i < r.length);
-
-    return r;
+    }
+    return possibleDeinflections;
   }
 
-  // katakana -> hiragana conversion tables
-  ch: number[] = [
-    0x3092, 0x3041, 0x3043, 0x3045, 0x3047, 0x3049, 0x3083, 0x3085, 0x3087,
-    0x3063, 0x30fc, 0x3042, 0x3044, 0x3046, 0x3048, 0x304a, 0x304b, 0x304d,
-    0x304f, 0x3051, 0x3053, 0x3055, 0x3057, 0x3059, 0x305b, 0x305d, 0x305f,
-    0x3061, 0x3064, 0x3066, 0x3068, 0x306a, 0x306b, 0x306c, 0x306d, 0x306e,
-    0x306f, 0x3072, 0x3075, 0x3078, 0x307b, 0x307e, 0x307f, 0x3080, 0x3081,
-    0x3082, 0x3084, 0x3086, 0x3088, 0x3089, 0x308a, 0x308b, 0x308c, 0x308d,
-    0x308f, 0x3093,
-  ];
+  isKana(charCode: number): boolean {
+    return (
+      (charCode >= KANA.HIRAGANA_START && charCode <= KANA.HIRAGANA_END) ||
+      (charCode >= KANA.KATAKANA_START && charCode <= KANA.KATAKANA_END) ||
+      (charCode >= KANA.HW_KATAKANA_START && charCode <= KANA.HW_KATAKANA_END)
+    );
+  }
 
-  cv: number[] = [
-    0x30f4, 0xff74, 0xff75, 0x304c, 0x304e, 0x3050, 0x3052, 0x3054, 0x3056,
-    0x3058, 0x305a, 0x305c, 0x305e, 0x3060, 0x3062, 0x3065, 0x3067, 0x3069,
-    0xff85, 0xff86, 0xff87, 0xff88, 0xff89, 0x3070, 0x3073, 0x3076, 0x3079,
-    0x307c,
-  ];
-  cs: number[] = [0x3071, 0x3074, 0x3077, 0x307a, 0x307d];
+  convertToHiragana(str: string): string {
+    let result = '';
+    for (let i = 0; i < str.length; i++) {
+      const char = str.charAt(i);
+      const nextChar = i + 1 <= str.length - 1 ? str.charAt(i + 1) : null;
+      const nextCharCode = nextChar?.charCodeAt(0);
+      const isSemiVoiced = nextCharCode === PUNCTUATION.SEMI_VOICED_MARK;
+      const isVoiced = nextCharCode === PUNCTUATION.VOICED_MARK;
+
+      const key = isSemiVoiced
+        ? char + nextChar
+        : isVoiced
+        ? char + nextChar
+        : char;
+      const hiragana = kanaToHiraganaNormalizationMap[key];
+      result += hiragana !== undefined ? hiragana : char;
```

Comment:
In this case, since we're already handling undefined can we make the lookup table smaller so that it only contains special characters we have to remap? That would make it more obvious which cases are special!